### ISSUE TITLE
mario slows down after a while

### ISSUE THREAD

**Issue by bburdette** [michaelficarra/purescript-demo-mario#11]
Have you noticed that Mario tends to slow down after a while?  I think its just when you're moving him around, the slowdown may not happen if he's just standing there without user input.  Gave it a try on firefox 38.0.5 and on chromium 41.0.2272.89 (64-bit).

**Comment by michaelficarra** [comment]
No, I've not noticed that. I'll try out a long running Mario game again. I've tested this previously, but that was before the 0.7 upgrade.

**Comment by bburdette** [comment]
because I only have psc 0.6.9.5 I used the older code ("workshop 0.6"), but the problem appears to manifest on the link at the top of the project page, which I assume is built with 0.7.

**Comment by michaelficarra** [comment]
Confirmed. I'll have to profile this to see why it is happening now. It was not happening with the 0.6.x version, and I'm not sure if it's something I changed or something that changed with the compiler.

**Comment by Fresheyeball** [comment]
I saw this in 0.6.x as well.

**Comment by Fresheyeball** [comment]
I profile this and right now it looks like its `__do()` is being used recursively on frame updates, and it leaks memory. Its GC is not cleaning up `__do()` function objects.

**Comment by RoganMurley** [comment]
Was any progress ever made on this?

**Comment by michaelficarra** [comment]
Not on my part. I'll have to update it for 0.9.x at some point and see if the problem persists.

**Comment by RoganMurley** [comment]
Ah, thank you. Do you think it is a problem with the compiler?

I'm writing a game and experiencing the same problem.
